components_library/notes.md

```diff
@@ -0,0 +1,62 @@
+# Notes setting up RISC-V FS in components
+
+## Thoughts
+
+I'm going to try to not use SimpleBoard and see what happens.
+...
+Nope, that was a bad idea.
+SimpleBoard's `connectThings` is useful and so it the Clock setup in the constructor.
+
+Setting up the I/O stuff is pretty awkward.
+For RISC-V it's going to be spread between the `RiscvBoard` and the `HiFive` platform.
+
+Also, setting up the I/O stuff is quite strange.
+I'm not sure what should be done by the board and what is done in the caches.
+
+## Things to do
+
+- Figure out a better solution for the dtb generation
+  - Make is so that we don't have to use the variable name "disk" for the platform device
+- Check if it works with no_cache. It should as the fs_linux.py file seems to.
+
+- MMU problems
+  - Need to have MMU caches when there are caches
+  - Or, if there are no caches, then need to have another bus to connect MMUs
+
+## General components improvements
+
+- Add a "CompatibilityError" instead of "EnvironmentError"
```

Comment:
Yes, i think someone should sit down and create some bespoke gem5 Errors. I've not been thinking very hard about this and generally throwing rather generic exceptions. There's a Jira ticket for this.